reference_controllers/rabbitmq_controller_unverified/src/main.rs

```diff
@@ -0,0 +1,340 @@
+            _ => return Err(Error::ConfigMapCreationFailed(e)),
+        },
+        _ => {}
+    }
+
+    // Create sever config
+    let server_config = server_configmap::server_configmap_build(&rabbitmq);
+    info!(
+        "Create server config: {}",
+        server_config.metadata.name.as_ref().unwrap()
+    );
+    match cm_api.create(&PostParams::default(), &server_config).await {
+        Err(e) => match e {
+            kube_client::Error::Api(kube_core::ErrorResponse { ref reason, .. })
+                if reason.clone() == "AlreadyExists" => {}
+            _ => return Err(Error::ConfigMapCreationFailed(e)),
+        },
+        _ => {}
+    }
+
+    // Create service account
+    let service_account = service_account::service_account_build(&rabbitmq);
+    info!(
+        "Create service account: {}",
+        service_account.metadata.name.as_ref().unwrap()
+    );
+    match svc_acc_api
+        .create(&PostParams::default(), &service_account)
+        .await
+    {
+        Err(e) => match e {
+            kube_client::Error::Api(kube_core::ErrorResponse { ref reason, .. })
+                if reason.clone() == "AlreadyExists" => {}
+            _ => return Err(Error::ServiceAccountCreationFailed(e)),
+        },
+        _ => {}
+    }
+
+    // Create role
+    let role = role::role_build(&rabbitmq);
+    info!("Create role: {}", role.metadata.name.as_ref().unwrap());
+    match role_api.create(&PostParams::default(), &role).await {
+        Err(e) => match e {
+            kube_client::Error::Api(kube_core::ErrorResponse { ref reason, .. })
+                if reason.clone() == "AlreadyExists" => {}
+            _ => return Err(Error::RoleCreationFailed(e)),
+        },
+        _ => {}
+    }
+
+    // Create role binding
+    let role_binding = role_binding::role_binding_build(&rabbitmq);
+    info!(
+        "Create role binding: {}",
+        role_binding.metadata.name.as_ref().unwrap()
+    );
+    match role_binding_api
+        .create(&PostParams::default(), &role_binding)
+        .await
+    {
+        Err(e) => match e {
+            kube_client::Error::Api(kube_core::ErrorResponse { ref reason, .. })
+                if reason.clone() == "AlreadyExists" => {}
+            _ => return Err(Error::RoleBindingCreationFailed(e)),
+        },
+        _ => {}
+    }
+
+    // Create statefulset
+    let statefulset = statefulset::statefulset_build(&rabbitmq);
+    info!(
+        "Create statefulset: {}",
+        statefulset.metadata.name.as_ref().unwrap()
+    );
+    match sts_api.create(&PostParams::default(), &statefulset).await {
+        Err(e) => match e {
+            kube_client::Error::Api(kube_core::ErrorResponse { ref reason, .. })
+                if reason.clone() == "AlreadyExists" => {}
+            _ => return Err(Error::StatefulSetCreationFailed(e)),
+        },
+        _ => {}
+    }
+
+    Ok(Action::requeue(Duration::from_secs(300)))
+}
+
+async fn reconcile_operator_defaults(
```

Comment:
It seems to me you are trying to update some empty value in your CR to some default value in this method? I see you are using the replace api, but let's make it simpler: can you either make the image hardcoded in your code, or make the image non-optional in your cr.spec so people have to specify it.